bootloader/main.c

```diff
@@ -204,7 +204,12 @@ patch_app(const char *prog_path)
 static char *
 create_tmpdir(void)
 {
-    static char template[] = "/tmp/staticx-XXXXXX";
+    const char* tmpdir_env = getenv("TMPDIR");
+    size_t tmpdir_size;
+    tmpdir_size = snprintf(NULL, 0, "%s/%s", (tmpdir_env!=NULL) ? tmpdir_env : "/tmp", "staticx-XXXXXX");
+    static char *template;
+    template = (char *)malloc(tmpdir_size +1);
```

Comment:
I suggest allocating an 1K-array on the stack. The malloc without free make me feel bad. This will simplify the code to:
```C
char template[1024];
size_t tmpsize = snprintf(template, sizeof(template), "%s/%s", (tmpdir_env!=NULL) ? tmpdir_env : "/tmp", "staticx-XXXXXX");
if(tmpsize == 1024) //Cause error
```
Also template should not be `static`.

Reply:
Thanks for the review. In #101 I actually used the existing `path_join` function to simplify this, and `free`d the result where appropriate.